padertorch/train/hooks.py

```diff
@@ -812,105 +816,136 @@ class StopTraining(Exception):
+        return trainer.loss_weights[self.name]
 
-        Args:
-            name: name of the attribute. You can use "attr1.attr11" to
-                anneal a sub attribute
-            factor: factor by which to anneal the attribute.
-                factor > 1. results in an increase while factor < 1. results
-                in a decrease
-            trigger:
-            max_value: upper bound of the weight
-            min_value: lower bound of the weight
-                (hint: can also be used to activate a loss weight after a
-                certain number of iterations/epochs)
-        """
-        super().__init__(trigger)
-        self.name = name.split('.')
-        assert (factor is None) ^ (slope is None), (factor, slope)
-        self.factor = factor
-        self.slope = slope
-        self.max_value = max_value
-        self.min_value = min_value
-        self.onset_value = None
+    def set_value(self, trainer, value):
+        trainer.loss_weights[self.name] = value
 
-    @property
-    def uid(self):
-        return super().uid + f"({'.'.join(self.name)})"
+
+class ModelAttributeAnnealingHook(AnnealingHook):
+    """
+    Anneals an (sub)attribute of the trainers model.
+    """
 
     def get_module(self, trainer: 'pt.Trainer'):
         module = trainer.model
-        for attr_name in self.name[:-1]:
+        name = self.name.split('.')[:-1]
+        for attr_name in name:
             module = getattr(module, attr_name)
         return module
 
-    def pre_step(self, trainer):
-        if self.trigger(iteration=trainer.iteration, epoch=trainer.epoch) \
-                and trainer.iteration != 0:
-            module = self.get_module(trainer)
-            value = getattr(module, self.name[-1])
-            if self.onset_value is None:
-                self.onset_value = value
-            if self.factor is not None:
-                value *= self.factor
-            if self.slope is not None:
-                value = self.onset_value + self.slope * trainer.iteration
-            if self.max_value is not None:
-                value = min(value, self.max_value)
-            if self.min_value is not None:
-                value = max(value, self.min_value)
-            setattr(module, self.name[-1], value)
+    def get_value(self, trainer):
+        module = self.get_module(trainer)
+        attr_name = self.name.split('.')[-1]
+        return getattr(module, attr_name)
 
-    def close(self, trainer):
-        if self.onset_value is not None:
-            module = self.get_module(trainer)
-            setattr(module, self.name[-1], self.onset_value)
+    def set_value(self, trainer, value):
+        module = self.get_module(trainer)
+        attr_name = self.name.split('.')[-1]
+        setattr(module, attr_name, value)
+
+
+class LRAnnealingHook(AnnealingHook):
+    """
+    Anneals an optimizer learning rate.
+    """
+    def __init__(self, trigger, breakpoints, name=None):
+        """
 
+        Args:
```

Comment:
Can you either copy the docstring from AnnealingHook or mention that the signature description is in AnnealingHook?